app/forms/instance-create.tsx

```diff
@@ -573,24 +588,97 @@ export function CreateInstanceForm() {
   )
 }
 
+// `ip is …` guard is necessary until we upgrade to 5.5, which handles this automatically
+const isFloating = (
+  ip: ExternalIpCreate
+): ip is { type: 'floating'; floatingIp: NameOrId } => ip.type === 'floating'
+
 const AdvancedAccordion = ({
   control,
   isSubmitting,
+  floatingIpList,
   siloPools,
 }: {
   control: Control<InstanceCreateInput>
   isSubmitting: boolean
+  floatingIpList: { items: Array<FloatingIp> }
   siloPools: Array<{ name: string; isDefault: boolean }>
 }) => {
   // we track this state manually for the sole reason that we need to be able to
   // tell, inside AccordionItem, when an accordion is opened so we can scroll its
   // contents into view
   const [openItems, setOpenItems] = useState<string[]>([])
+  const [selectedFloatingIp, setSelectedFloatingIp] = useState<string | undefined>()
   const externalIps = useController({ control, name: 'externalIps' })
   const ephemeralIp = externalIps.field.value?.find((ip) => ip.type === 'ephemeral')
   const assignEphemeralIp = !!ephemeralIp
   const selectedPool = ephemeralIp && 'pool' in ephemeralIp ? ephemeralIp.pool : undefined
   const defaultPool = siloPools.find((pool) => pool.isDefault)?.name
+  const attachedFloatingIps = (externalIps.field.value || []).filter(isFloating)
+
+  // Filter out the IPs that are already attached to an instance
+  const attachableFloatingIps = useMemo(
+    () => floatingIpList.items.filter((ip) => !ip.instanceId),
+    [floatingIpList]
+  )
+
+  // To find available floating IPs, we remove the ones that are already committed to this instance
+  const availableFloatingIps = attachableFloatingIps.filter(
+    (ip) => !attachedFloatingIps.find((attachedIp) => attachedIp.floatingIp === ip.name)
+  )
+  const attachedFloatingIpsData = attachedFloatingIps
+    .map((ip) => attachableFloatingIps.find((fip) => fip.name === ip.floatingIp))
+    .filter((x) => !!x) as Array<FloatingIp>
+
+  const addFloatingIpPlaceholder = () => {
+    setSelectedFloatingIp(undefined)
+    const newExternalIps = [
+      ...(externalIps.field.value || []),
+      { type: 'floating', floatingIp: '' },
+    ]
+    externalIps.field.onChange(newExternalIps)
+  }
+
+  const removeFloatingIpPlaceholder = () => {
+    externalIps.field.onChange(
+      externalIps.field.value?.filter(
+        (ip) => ip.type === 'ephemeral' || ip.floatingIp !== ''
+      )
+    )
+  }
+
+  const attachFloatingIp = (name: string) => {
+    const newExternalIps = externalIps.field.value?.map((ip) => {
+      if (ip.type === 'floating' && ip.floatingIp === '') {
+        return { type: 'floating', floatingIp: name }
+      }
+      return ip
+    })
+    externalIps.field.onChange(newExternalIps)
+  }
```

Comment:
This also becomes simpler without the placeholder concept: you just add `{ type: 'floating', floatingIp: name }` to the list.